openapi-core/src/main/java/io/ballerina/openapi/core/generators/common/OASModifier.java

```diff
@@ -53,40 +54,54 @@
  * @since 2.1.0
  */
 public class OASModifier {
-    OpenAPI openapi;
     List<Diagnostic> diagnostics = new ArrayList<>();
 
-    public OASModifier(OpenAPI openapi) {
-        this.openapi = openapi;
-    }
 
     public List<Diagnostic> getDiagnostics() {
         return diagnostics;
     }
 
-    //modifyWithBallerinaConventions 
-    public OpenAPI modifyWithBallerinaConventions() {
+    public Optional<Map<String, String>> getProposedNameList(OpenAPI openapi) {
+        Map<String, String> nameMap = new HashMap<>();
+        if (openapi.getComponents() == null) {
+            return Optional.empty();
+        }
+        Components components = openapi.getComponents();
+        Map<String, Schema> schemas = components.getSchemas();
+        if (schemas == null) {
+            return Optional.empty();
+        }
+        Set<String> modifiedSchemaNames = new HashSet<>();
+
+        for (Map.Entry<String, Schema> schemaEntry: schemas.entrySet()) {
+            //modified logic should come to here
+            String schemaName = schemaEntry.getKey();
+            String modifiedName = getValidNameForType(schemaEntry.getKey());
+            if (!schemaName.equals(modifiedName) && (schemas.containsKey(modifiedName) ||
```

Comment:
Lets remove this check for duplication from the `getProposedNameList` method since this will be provided as a separate function where the input is the initial name map and the output is a resolved name map

Reply:
Please use the following function to get conflict resolved names:
```java
public static Map<String, String> getResolvedNameMap(Map<String, String> nameMap) {
    Map<String, String> resolvedNames = new HashMap<>();
    Map<String, Integer> nameCount = new HashMap<>();

    for (Map.Entry<String, String> entry : nameMap.entrySet()) {
        String currentName = entry.getKey();
        String newName = entry.getValue();
        String resolvedName = newName;

        while (resolvedNames.containsValue(resolvedName)) {
            int count = nameCount.getOrDefault(newName, 1);
            resolvedName = newName + count;
            nameCount.put(newName, count + 1);
        }

        resolvedNames.put(currentName, resolvedName);
    }

    return resolvedNames;
}
```

Reply:
I will update